src/System.Text.Primitives/System/Text/Formatting/FloatFormatter.cs

```diff
@@ -44,7 +44,42 @@ public static bool TryFormatNumber(double value, bool isSingle, Span<byte> buffe
             // TODO: the lines below need to be replaced with properly implemented algorithm
             // the problem is the algorithm is complex, so I am commiting a stub for now
             var hack = value.ToString(format.Symbol.ToString());
-            return encoder.TryEncode(hack, buffer, out bytesWritten);
+            var utf16Bytes = hack.AsSpan().AsBytes();
+            if (symbolTable == SymbolTable.InvariantUtf8)
+            {
+                var status = Encoders.Utf8.ConvertFromUtf16(utf16Bytes, buffer, out int consumed, out bytesWritten);
+                return status == Buffers.TransformationStatus.Done;
+            }
+            else if (symbolTable == SymbolTable.InvariantUtf16)
+            {
+                bytesWritten = utf16Bytes.Length;
+                if (utf16Bytes.TryCopyTo(buffer))
+                    return true;
+
+                bytesWritten = 0;
+                return false;
+            }
+            else
+            {
+                // TODO: This is currently pretty expensive. Can this be done more efficiently?
+                //       Note: removing the hack might solve this problem a very different way.
+                var status = Encoders.Utf8.ComputeEncodedBytesFromUtf16(utf16Bytes, out int needed);
+                if (status != Buffers.TransformationStatus.Done)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+
+                Span<byte> temp = new byte[needed];
```

Comment:
Good call. Given this is guaranteed to be small (under 64 bytes), I made that change.